src/handler.ts

```diff
@@ -1,14 +1,104 @@
-import 'source-map-support/register';
+import { Handler, SQSBatchResponse, SQSEvent } from 'aws-lambda';
+import { InvokeCommand, LambdaClient } from '@aws-sdk/client-lambda';
+import { Readable } from 'stream';
+import { Blob } from 'buffer';
+import { PutObjectCommand, S3 } from '@aws-sdk/client-s3';
 import logger from './observability/logger';
+import { generateMinistryDocumentModel } from './models/document';
+import { Request } from './models/request';
+import { DocumentType } from './models/documentType.enum';
 
-const { NODE_ENV, SERVICE, AWS_PROVIDER_REGION, AWS_PROVIDER_STAGE } = process.env;
+const handler: Handler = async (event: SQSEvent): Promise<SQSBatchResponse> => {
+  logger.debug("Function triggered'.");
+  if (!event || !event.Records || !event.Records.length) {
+    logger.error('ERROR: event is not defined.');
+    throw new Error('Event is empty');
+  }
 
-logger.info(
-  `\nRunning Service:\n '${SERVICE}'\n mode: ${NODE_ENV}\n stage: '${AWS_PROVIDER_STAGE}'\n region: '${AWS_PROVIDER_REGION}'\n\n`,
-);
+  const altPromiseArray = event.Records.map((sqsRecord) => {
+    const request = JSON.parse(sqsRecord.body) as Request;
+    let documentData;
+    let fileName: string;
+    if (request.documentName === DocumentType.MINISTRY || request.documentName === DocumentType.MINITRY_TRL) {
+      documentData = generateMinistryDocumentModel(request.vehicle);
+      fileName = request.vehicle.techRecord[0].plates.plateSerialNumber;
+    } else {
+      throw new Error('Document Type not supported');
+    }
+    return generateAndUpload(documentData, request, fileName);
+  });
 
-const handler = async (): void => {
-  logger.debug("Function triggered'.");
+  const results = await Promise.allSettled(altPromiseArray);
+  const ids = results
+    .map((result, index) => (result.status === 'fulfilled' ? null : event.Records[index].messageId))
+    .filter((item) => item !== null);
+  return {
+    batchItemFailures: ids.map((id) => ({ itemIdentifier: id })),
+  };
+};
+
+const generateAndUpload = async (documentData, request: Request, fileName: string) => {
```

Comment:
can we look at moving these into util or service folders? ideally would like just the handler in the handler file right?